### DISCUSSION TITLE
On-demand feature generation

### DISCUSSION THREAD

**Discussion by mikeydavison** [feathr-ai/feathr#106] [Q&A] [answered]
Hello.  Does Feathr support on-demand feature generation?  From the docs and examples it is clear how batch and online queries work, but I couldn't find anything that enables on-demand generation of features (e.g., feature data comes not from online store but from API call).  Apologies if I've missed this in the docs.

**Comment by hangfei** [comment, marked as answer]
It's not supported right now in the opne-source branch. But we do have this functionality inside LinkedIn and we will roll it out in a few months.

If you have source data that is streamed into your data lake/warehouse, you can use streaming feature generation which will come in a few weeks.

If you can elaborate a bit more on your use case, that will be better. Like the shape of your data, what's the API client to access those endpoints, do you need any kind of transformation on top of the raw data, what's the latency requirements?

Thanks.

**Reply by mikeydavison** [reply]
Thanks!  The scenario I'm considering is inferencing via an API with a JSON payload of some sort.  In this scenario the payload doesn't correspond to features in the online or offline stores, but I'd like to be able to apply the same transformations as were applied to those stores so as to ensure consistency.  It sounds like this capability exists and is coming in the near future though.  Thanks again.

**Reply by JnsLns** [reply]
Hi! Is this still valid? Or has the capability been added in the meantime? 
Best Regards!
